### DISCUSSION TITLE
Seeking guidance on standard

### DISCUSSION THREAD

**Discussion by dgreene1** [microsoft/RulesEngine#415] [General]
Can you share with me if there is a universally agreed upon standard for sending rules over HTTP via JSON?

Because I notice that there is a schema[a schema in the repo](https://github.com/microsoft/RulesEngine/blob/a74f73a44e38ce27c76945f15fd697c7abe5eef9/schema/workflow-schema.json#L1) but it doesn’t appear to be a standard— it’s just what this library uses.

So what I’m looking for is a JSON Schema spec that I can point to to say “this is what the API that reads the rule selections expects to receive.”

For full context, my team is building a rule builder UI and unsure of what engine we’ll be using. So having a standardized output will allow us to make that swap easily (ie the adapter pattern). And it will let us start building the UI before the engine has been decided upon. Thank you for the guidance.

Side note: I find this library to be wonderful but I have concerns about the expression part of the schema since it seems unclear what would constitute a valid string. Like is it any JS?

**Comment by abbasc52** [comment]
Hi @dgreene1, currently there is no universally agreed upon standard. 

On your second question, the expression used in RulesEngine is C# expression . You can find more details here - https://microsoft.github.io/RulesEngine/#c-expression-support

**Comment by YogeshPraj** [comment]
Most widely use format to configure rules from UI. 

![image](https://user-images.githubusercontent.com/41417278/195816296-8a10768f-1659-4e6c-bf21-095e959da5d4.png)

This requires simple json structure to store values from UI.
![image](https://user-images.githubusercontent.com/41417278/195816803-bd757eb7-1060-41b5-a482-004d8048cb24.png)

Backend can use this format to translate to respective BRE engine format using adapter pattern (e.g, MicrosoftBRETranslatorPlugin).
